src/utilsplitstring.h

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) 2018 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <iterator>
+#include <set>
+#include <string>
+#include <vector>
+#include <type_traits>
+
+#ifndef BITCOIN_UTILSPLITSTRING_H
+#define BITCOIN_UTILSPLITSTRING_H
+
+/**
+ * Tokenizes a string by any of the given separators.
+ * @param[in] tokens            The container (either an instance of std::vector<std::string>
+ *                              or std::set<std::string>) to add tokenized string parts to.
+ * @param[in] str               The string to tokenize.
+ * @param[in] anyOfSeparator    A string with valid separators.
+ * @param[in] mergeEmpty        Set to true to merge adjacent separators (empty tokens); otherwise false (default).
+ */
+template<typename ContainerT>
+void Split(ContainerT& tokens, const std::string& str, const std::string& anyOfSeparator, bool mergeEmpty = false)
+{
+    static_assert(
+        std::is_same<std::vector<std::string>, ContainerT>::value ||
+        std::is_same<std::set<std::string>, ContainerT>::value,
+        "ContainerT must be of type std::vector<std::string> or std::set<std::string>");
```

Comment:
I'm not sure this restriction is helpful - absent it the args would be limited to whatever is compatible with `std::inserter`, which could be fine...

Reply:
I did add the `static_assert`, because there were no other use cases in the codebase and didn't want to make any assumptions for containers other than `std::vector` and `std::set`. That said, you've raised a valid point and I am happy to remove it.